pyiron_nodes/atomistic/engine/vasp.py

```diff
@@ -0,0 +1,263 @@
+import shutil
+import subprocess
+from typing import Optional, List, Dict, Tuple
+from dataclasses import dataclass, field
+import pandas as pd
+from pyiron_workflow import Workflow
+from pyiron_atomistics.vasp.output import parse_vasp_output as pvo
+from pymatgen.core import Structure
+from pymatgen.io.vasp.inputs import Incar, Kpoints
+from pymatgen.io.vasp.outputs import Vasprun
+from ase import Atoms
+
+POTCAR_library_path = "/cmmc/u/hmai/vasp_potentials_54/"
+POTCAR_specification_data = str(
+    Path(__file__).parent.joinpath("vasp_pseudopotential_PBE_data.csv")
+)
+
+
+@dataclass
+class VaspInput:
+    structure: Structure
+    incar: Incar
+    pseudopot_lib_path: str = field(default=POTCAR_library_path)
+    potcar_paths: Optional[List[str]] = None
+    kpoints: Optional[Kpoints] = None
+
+
+def is_line_in_file(filepath: str, line: str, exact_match: bool = True) -> bool:
+    try:
+        with open(filepath, "r") as file:
+            for file_line in file:
+                if exact_match and line == file_line.strip():
+                    return True
+                elif not exact_match and line in file_line:
+                    return True
+    except FileNotFoundError:
+        print(f"File '{filepath}' not found.")
+    return False
+
+
+def write_POSCAR(workdir: str, structure: Structure, filename: str = "POSCAR") -> str:
+    poscar_path = os.path.join(workdir, filename)
+    structure.to(fmt="poscar", filename=poscar_path)
+    return poscar_path
+
+
+def write_INCAR(workdir: str, incar: Incar, filename: str = "INCAR") -> str:
+    incar_path = os.path.join(workdir, filename)
+    incar.write_file(incar_path)
+    return incar_path
+
+
+def write_POTCAR(workdir: str, vasp_input, filename: str = "POTCAR") -> str:
+    class PotcarNotGeneratedError(Exception):
+        pass
+
+    if vasp_input.potcar_paths is None:
+        potcar_paths = get_default_POTCAR_paths(
+            vasp_input.structure, vasp_input.pseudopot_lib_path
+        )
+    else:
+        potcar_paths = vasp_input.potcar_paths
+
+    potcar_path = os.path.join(workdir, filename)
+
+    with open(potcar_path, "wb") as wfd:
+        for f in potcar_paths:
+            with open(f, "rb") as fd:
+                shutil.copyfileobj(fd, wfd)
+
+    return potcar_path
+
+
+def write_KPOINTS(
+    workdir: str, kpoints: Optional[Kpoints] = None, filename: str = "KPOINTS"
+) -> str:
+    kpoint_path = os.path.join(workdir, filename)
+    if kpoints is not None:
+        kpoints.write_file(kpoint_path)
+    else:
+        with open(kpoint_path, "w") as f:
+            f.write(
+                "Automatic mesh\n0\nGamma\n1 1 1\n0 0 0\n"
+            )  # Example KPOINTS content
+    return kpoint_path
+
+
+@Workflow.wrap.as_function_node("workdir")
+def create_WorkingDirectory(workdir: str, quiet: bool = False) -> str:
+    # Check if workdir exists
+    if not os.path.exists(workdir):
+        os.makedirs(workdir)
+        print(f"made directory '{workdir}'")
```

Comment:
At this stage I don't have super strong feelings about it, but I think I lean towards replacing all the print statements with `logger.log(...)`, where you can `from pyiron_workflow import logger` (which in turn is just using the `pyiron_snippets` logger functionality). One can imagine a similar replacement with `warnings.warn` and `logger.warning`, but that is actually kind of a significant change to functionality and I am not yet convinced of it.